cas-server-core-authentication/src/main/java/org/apereo/cas/authentication/CurrentCredentialsAndAuthentication.java

```diff
@@ -0,0 +1,93 @@
+package org.apereo.cas.authentication;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static java.util.stream.Collectors.joining;
+
+/**
+ * ThreadLocal based holder for current set of credentials and/or authentication object for any current
+ * CAS authentication transaction. Useful for making this information available to all the interested CAS
+ * components that are not tightly coupled with core CAS APIs, for example audit principal resolver component, etc.
+ * <p>
+ * The thread local state carried by this class should be set by core CAS components processing core authentication and
+ * CAS protocol events e.g. <code>AbstractAuthenticationManager</code>, <code>CentralAuthenticationServiceImpl</code>, etc.
```

Comment:
Minor suggestion: try using {@link}. Helps with future refactoring, if the class should be renamed, etc.